src/rust/engine/fs/src/lib.rs

```diff
@@ -398,7 +395,12 @@ impl PosixFS {
     Ok(stats)
   }
 
-  fn pool(&self) -> RwLockReadGuard<Option<CpuPool>> {
+  // TODO: This probably shouldn't be public.
+  // This is called from in nodes.rs's Snapshot::create.
+  // How should a node do this kind of rate-limiting? It seems weird for a node to be passing a
+  // PosixFS into fs::Snapshot::from_path_stats just so it can get at its pool.
+  // So I guess we need to move this pool somewhere else. But where? Maybe Context?
```

Comment:
Touched on this on #5105, but: I think what we'll need is an Arc to share the pool between the various members of the `Core` that need it, including `PosixFS` and `Store` (which isn't using it yet, but should be to take IO off the main thread). Opened https://github.com/pantsbuild/pants/issues/5108 about this.

But we also need the fork-safe machinery to ensure that we recreate this when necessary. So I think it will look like an `Arc<ForksafePool>`, with a helper `fn with_pool<F: FnOnce()->T>(f: F) -> T`.

Reply:
Implemented here: #5110